authzed/api/v1/error_reason.proto

```diff
@@ -134,9 +134,26 @@ enum ErrorReason {
   //     { "reason": "ERROR_REASON_INVALID_SUBJECT_TYPE",
   //       "domain": "authzed.com",
   //       "metadata": {
+  //         "definition_name": "somedefinition",
   //         "relation_name": "somerelation",
   //         "subject_type": "user:*"
   //       }
   //     }
   ERROR_REASON_INVALID_SUBJECT_TYPE = 10;
+
+  // The request tries to specify a caveat parameter value with the wrong type.
+  //
+  // Example of an ErrorInfo:
+  //
+  //     { "reason": "ERROR_REASON_CAVEAT_PARAMETER_TYPE_ERROR",
+  //       "domain": "authzed.com",
+  //       "metadata": {
+  //         "definition_name": "somedefinition",
+  //         "relation_name": "somerelation",
+  //         "caveat_name": "somecaveat",
+  //         "parameter_name": "someparameter",
+  //         "expected_type": "int",
+  //       }
+  //     }
+  ERROR_REASON_CAVEAT_PARAMETER_TYPE_ERROR = 11;
```

Comment:
Does it make sense to have one error type for type errors and just include the additional details in the metadata?

Reply:
IMO, no. In this case: it is a type error with the *usage* of the caveat, rather than the *definition* of the schema. It will have different fields, and different semantics

Reply:
Isn't this similar to the `ERROR_REASON_INVALID_SUBJECT_TYPE` above it which is also an error with the usage of a subject rather than the definition?

Reply:
No; the subject *type* is valid, but the context parameter is not; they are different errors with different metadata returned